stdlib/ballerina-http/src/main/java/org/ballerinalang/net/uri/URIUtil.java

```diff
@@ -39,4 +44,35 @@ public static String getSubPath(String path, String basePath) {
 
         return path.substring(basePath.length());
     }
+
+    public static String removeMatrixParams(String path, Map<String, Map<String, String>> matrixParams) {
+        if (path.startsWith("/")) {
+            path = path.substring(1);
+        }
+        String[] pathSplits = path.split("\\?");
+        String[] pathSegments = pathSplits[0].split("/");
+        String pathToMatrixParam = "";
+        for (String pathSegment : pathSegments) {
+            String[] splitPathSegment = pathSegment.split(";");
+            pathToMatrixParam = pathToMatrixParam.concat("/" + splitPathSegment[0]);
+            Map<String, String> segmentMatrixParams = new HashMap<>();
+            if (splitPathSegment.length > 1) {
+                for (int i = 1; i < splitPathSegment.length; i++) {
+                    String[] splitMatrixParam = splitPathSegment[i].split("=");
+                    if (splitMatrixParam.length != 2) {
```

Comment:
Shouldn't we check malformed paths such as
- /mercedes/e55;color=/2006
- /mercedes/e55;=black/2006/
in above 2nd case, splitMatrixParam.length == 2 but splitMatrixParam[0] is empty. So we need to check that too

Reply:
IMO we should not handle it our selves. If a client sends a request with such a URI which is wrong is an issue with that client. If we are not rejecting the request IMHO there is no point of checking it at this time since this is a time critical operation :)